### PR TITLE
tests: partition_manager: Add static_pm_file test

### PR DESCRIPTION
**Pull request by nordicjm**
Adds a test that checks that PM_STATIC_YML_FILE functionality is working

### CODE REVIEW

**Review by katgiadla — APPROVED** [review]
Please add responsible person for these tests in `CODEOWNERS` file.

**Review by katgiadla — CHANGES_REQUESTED** [review]
Please add responsible person for these tests in CODEOWNERS file.

File: CODEOWNERS [hunk 1]
```diff
@@ -358,6 +358,8 @@ Kconfig*                                  @tejlmand
 /tests/subsys/net/lib/wifi_credentials*/  @nrfconnect/ncs-cia
 /tests/subsys/net/lib/mqtt_helper/        @nrfconnect/ncs-cia
 /tests/subsys/partition_manager/region/   @hakonfam @sigvartmh
+/tests/subsys/partition_manager/region/   @hakonfam @sigvartmh
```
Comment (katgiadla) [inline CODEOWNERS:361]: Duplicate with line 360. Please remove this line

**Review by katgiadla — CHANGES_REQUESTED** [review]

**Review by katgiadla — APPROVED** [review]
LGTM

**Review by sigvartmh — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**